### PR TITLE
Not giving diagnostic message on unknown types being bases

### LINKED ISSUE #1361: False positives for "inheriting from something that is not a class"
The Diagnostics for Inheriting from something that is not a class reports false positives when a library like [PonyORM](https://github.com/ponyorm/pony) creates classes dynamically.

MWE:
```python
from pony.orm import *
db = Database()
# You're supposed to have your entities inherit from db.Entity
# For reference, this gets created in pony/orm/core.py:746
class MyEntity(db.Entity): # Diagnostic warning because the python language server thinks we're not inheriting from a class
    mystr = Optional(str)

db.bind(provider='sqlite', filename='database.sqlite', create_db=True, create_tables=True)
```

### PR DESCRIPTION
**Pull request by CTrando**
Fixes #1361

### CODE REVIEW

File: src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs [hunk 1]
```diff
@@ -120,6 +120,8 @@ internal sealed class ClassEvaluator : MemberEvaluator {
                 var m = Eval.GetValueFromExpression(expr);
 
                 switch (m) {
+                    // Allow unknown members 
+                    case IMember member when member.GetPythonType() == Eval.UnknownType:
```
Comment (jakebailey) [inline src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]: Why `when member.GetPythonType() == Eval.UnknownType` over `when member.IsUnknown()`?

Comment (MikhailArkhipov) [inline src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]:
```cs
if(member.IsUnknown()) {
    return ...
}
```

Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]: In fact, `m` is already an `IMember`, so you can just do `if (m.IsUnknown())` before the `switch`.

Comment (CTrando) [reply src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]: Ah man, I forgot about that extension method.

File: src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
```
Comment (jakebailey) [inline src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]: Remove?

Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]: (A few usings are unused, the document could use an autoformat)

Comment (CTrando) [reply src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs]: Ok.

**Pull request merged**